packages/foam-vscode/src/core/model/workspace.ts

```diff
@@ -107,44 +81,53 @@ export class FoamWorkspace implements IDisposable {
     }
   }
 
+  public listById(resourceId: string): Resource[] {
+    if (!hasExtension(resourceId)) {
+      resourceId = resourceId + '.md';
+    }
+    resourceId = normalize(resourceId);
+    let resources = [];
+    for (const key of this.resources.keys()) {
+      if (key.endsWith(resourceId)) {
+        resources.push(this.resources.get(normalize(key)));
+      }
+    }
+    return resources;
+  }
+
   public find(resourceId: URI | string, reference?: URI): Resource | null {
     const refType = getReferenceType(resourceId);
     switch (refType) {
       case 'uri':
         const uri = resourceId as URI;
         return this.exists(uri)
-          ? this.resources.get(uriToResourceId(uri)) ?? null
+          ? this.resources.get(normalize(uri.path)) ?? null
           : null;
 
       case 'key':
-        const name = pathToResourceName(resourceId as string);
-        let paths = this.resourcesByName.get(name);
-
-        if (isNone(paths) || paths.length === 0) {
-          paths = this.resourcesByName.get(resourceId as string);
-        }
-
-        if (isNone(paths) || paths.length === 0) {
-          return null;
-        }
-        // prettier-ignore
-        const sortedPaths = paths.length === 1
-          ? paths
-          : paths.sort((a, b) => a.localeCompare(b));
-
-        return this.resources.get(sortedPaths[0]) ?? null;
+        const resources = this.listById(resourceId as string);
+        const sorted = resources.sort((a, b) =>
+          a.uri.path.localeCompare(b.uri.path)
+        );
+        return sorted[0] ?? null;
 
       case 'absolute-path':
+        if (!hasExtension(resourceId as string)) {
+          resourceId = resourceId + '.md';
+        }
         const resourceUri = URI.file(resourceId as string);
-        return this.resources.get(uriToResourceId(resourceUri)) ?? null;
+        return this.resources.get(normalize(resourceUri.path)) ?? null;
 
       case 'relative-path':
         if (isNone(reference)) {
           return null;
         }
+        if (!hasExtension(resourceId as string)) {
+          resourceId = resourceId + '.md';
+        }
         const relativePath = resourceId as string;
         const targetUri = URI.computeRelativeURI(reference, relativePath);
```

Comment:
Above you used the name `resourceUri`, I'm not sure whether there is a difference in meaning here that justifies to call it otherwise.

Reply:
This is because JS is stupid ;)

Because I prefer to use `const` instead of `let` I need to find different names in the various `case` statements, or it will trigger a compiler error. 
If you use in this line `const resourceUri = URI.computeRelativeURI(reference, relativePath);`, you will see what I mean.

The alternative would be to define `let resourceUri` beforehand and they assign it, which is actually something I am doing now in https://github.com/foambubble/foam/pull/856 exactly for the reason above.